dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/http1/DefaultHttp11ServerTransportListener.java

```diff
@@ -82,6 +83,15 @@ protected HttpMessageListener buildHttpMessageListener() {
         return new DefaultHttpMessageListener(listeningDecoder);
     }
 
+    @Override
+    protected void doOnData(HttpInputMessage message) {
```

Comment:
Is http2 also need it? so I think add it into Ondata is more reasonable:
```Java
public void onData(MESSAGE message) {
        executor.execute(() -> {
            try {
                doOnData(message);
            } catch (Throwable t) {
                logError(t);
                onError(t);
            } finally {
                try {
                    message.close();
                } catch (Exception e) {
                    onError(e);
                }
            }
        });
    }
```

Reply:
This only applies to http1 and http2, not grpc, because grpc will accumulate inputStream.
![image](https://github.com/apache/dubbo/assets/18413695/e835b5d5-7045-4caa-a26f-cad0fa3de47d)

ByteBuf is released when an exception occurs in grpc's doOnData process. Under normal circumstances, ByteBuf will be released by the close method of LengthFieldStreamingDecoder.
Normal scene:
![image](https://github.com/apache/dubbo/assets/18413695/b625a697-fadf-47d2-9f24-acf655fbd008)
Error scene:
![image](https://github.com/apache/dubbo/assets/18413695/51bc2ff0-9819-407b-be54-ee41d116845a)